researchstack-mpower/src/main/java/org/sagebionetworks/research/mpower/researchstack/framework/MpTaskHelper.java

```diff
@@ -0,0 +1,231 @@
+ * 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+ * may be used to endorse or promote products derived from this software without
+ * specific prior written permission. No license is granted to the trademarks of
+ * the copyright holders even if such marks are included in this software.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.sagebionetworks.research.mpower.researchstack.framework;
+
+import android.content.Context;
+import android.support.annotation.Nullable;
+
+import org.joda.time.DateTime;
+import org.researchstack.backbone.AppPrefs;
+import org.researchstack.backbone.ResourceManager;
+import org.researchstack.backbone.answerformat.AnswerFormat;
+import org.researchstack.backbone.model.TaskModel;
+import org.researchstack.backbone.result.FileResult;
+import org.researchstack.backbone.result.Result;
+import org.researchstack.backbone.result.StepResult;
+import org.researchstack.backbone.storage.NotificationHelper;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.data.Archive;
+import org.sagebionetworks.bridge.data.ArchiveFile;
+import org.sagebionetworks.bridge.data.JsonArchiveFile;
+import org.sagebionetworks.bridge.researchstack.TaskHelper;
+import org.sagebionetworks.bridge.researchstack.factory.ArchiveFileFactory;
+import org.sagebionetworks.bridge.researchstack.survey.SurveyAnswer;
+import org.sagebionetworks.bridge.researchstack.wrapper.StorageAccessWrapper;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.MpSmartSurveyTask;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpBooleanAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpChoiceAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpIntegerAnswerFormat;
+import org.sagebionetworks.research.mpower.researchstack.framework.step.body.MpTextQuestionBody;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MpTaskHelper extends TaskHelper {
+
+    public static final String ANSWERS_FILENAME = "answers.json";
```

Comment:
Should be "answers" and *not* "answers.json"

In order to work with the rules for Upload Schemas, they are designed where the schema keys are as follows:

```
schemaKey = <filename> + "." + <json key>
```

So, for example, if your JSON looks like this:

```
{
    "foo" : "bar",
    "blu" : "goo"
}
````

And your filename is "answers", then your Upload Schema would have `answers.foo` and `answers.blu` as the keys in the schema table.

Note: This is *not* the archive structure for the AppCore compatible Surveys, just the Upload Schemas. So the question in the active tasks about medication timing and which hand is used for tapping needs to use the answer map format (b/c it uses an Upload Schema) but the Surveys still use the old format where each answer is in a separate file called "foo.json" with the previously defined structure.
 
**Additionally**, the profile is expecting to find a Report that uses the answer map format for the `clientData` that it can use to build the survey question/answer pairs to show in the profile.

Reply:
Everything you're saying here is how we implemented it in the BP app.  

First, I need to get access to Synapse to verify these are making it.  I'll sync with Alx tomorrow.

Second, a few days ago, when I ran the demographics survey mPower2 on iOS. I noticed that even though it was a survey, the archive was uploading "answers.json" as the filename (not just "answers") with the results all in one file, and not separate json files for each answer.  This was contrary to how I thought surveys worked, and so I just figured surveys V2 was now all in one "answers.json" file.  

I can easily switch it back to the previous all in one file survey answers, but I'm a little confused on why I was seeing that output on iOS.  I just ran demographics survey again to sanity check myself, and this was the output, what am I missing?

```
insertDictionaryIntoArchive:filename:createdOn:]): Archiving metadata.json: {
    appName = mPower2;
    appVersion = "version 2.0, build 21";
    dataGroups = "gr_BR_II,parkinsons,test_user,gr_BR_AD,gr_DT_T,test_no_consent";
    deviceInfo = "Simulator; iOS/11.3";
    deviceTypeIdentifier = "x86_64";
    endDate = "2018-10-17T21:57:01.474-04:00";
    files =     (
                {
            contentType = "application/json";
            filename = "answers.json";
            timestamp = "2018-10-17T21:57:01.785-04:00";
        }
    );
    rsdFrameworkVersion = "version 1.0, build 20";
    schemaIdentifier = Demographics;
    schemaRevision = 1;
    startDate = "2018-10-17T21:56:42.178-04:00";
    taskIdentifier = Demographics;
    taskRunUUID = "9EAAE076-E039-4327-A947-610E5CEEC25F";
}
2018-10-17 21:57:01.788132-0400 mPower2[70411:1336802] SBBDataArchive.m line 145 (-[SBBDataArchive insertDictionaryIntoArchive:filename:createdOn:]): Archiving info.json: {
    appName = mPower2;
    appVersion = "version 2.0, build 21";
    files =     (
                {
            contentType = "application/json";
            filename = answers;
            timestamp = "2018-10-17T21:57:01.785-04:00";
        },
                {
            contentType = "application/json";
            filename = "metadata.json";
            timestamp = "2018-10-17T21:56:42.178-04:00";
        }
    );
    item = Demographics;
    phoneInfo = "iPhone Simulator; iOS/11.3";
    schemaRevision = 1;
}
```

Reply:
Hmmm.... I need to check on that. I think that's a bug on iOS.

Reply:
So I just ran all the surveys and tasks and the uploads are here:

https://sagebionetworks.jira.com/browse/MPOW-175

The filename is "answers". I ran this against master.

Reply:
Thanks for linking that.  I re-enabled single answer json files for the MpTaskHelper so now it does as iOS and uploads the single files for surveys but also the answers.json